linebacker/src/main/scala/io/chrisdavenport/linebacker/contexts/Executors.scala

```diff
@@ -37,38 +46,81 @@ object Executors {
    * 1 as to maximize the other pool for handling
    * requests or other work.
    */
-  def workStealingPool[F[_]: Sync](n: Int) =
-    executorServiceResource(workStealingPoolUnsafe(n))
+  def workStealingPool[F[_]: Sync](n: Int): Resource[F, ExecutorService] =
+    workStealingPoolC(n)(Sync[F].point(ForkJoinPool.defaultForkJoinWorkerThreadFactory))
+
+  def workStealingPoolC[F[_]: Sync](n: Int)(
+      threadFactory: F[ForkJoinWorkerThreadFactory]): Resource[F, ExecutorService] =
+    executorServiceResource(workStealingPoolUnsafe(n, threadFactory))
 
   /**
    * Default Pool For Scala, optimized for forked work and then returning to a
    * main pool, generally ideal for your main event loop.
    */
-  def forkJoinPool[F[_]: Sync](n: Int) =
-    executorServiceResource(forkJoinPoolUnsafe(n))
+  def forkJoinPool[F[_]: Sync](n: Int): Resource[F, ExecutorService] =
+    forkJoinPoolC(n)(Sync[F].point(ForkJoinPool.defaultForkJoinWorkerThreadFactory))
+
+  def forkJoinPoolC[F[_]: Sync](n: Int)(
+      threadFactory: F[ForkJoinWorkerThreadFactory]): Resource[F, ExecutorService] =
+    executorServiceResource(forkJoinPoolUnsafe(n, threadFactory))
 
   private def executorServiceResource[F[_]: Sync](
       f: F[ExecutorService]): Resource[F, ExecutorService] =
     Resource.make[F, ExecutorService](f)(es => Sync[F].delay(es.shutdownNow).void)
 
   object unsafe {
-    def unboundExecutorUnsafe[F[_]: Sync]: F[ExecutorService] = Sync[F].delay {
-      E.newCachedThreadPool(new ThreadFactory {
-        private val counter = new AtomicLong(0L)
-
-        def newThread(r: Runnable) = {
-          val th = new Thread(r)
-          th.setName("linebacker-thread-" + counter.getAndIncrement.toString)
-          th.setDaemon(true)
-          th
+    private[Executors] def unboundThreadFactory[F[_]](implicit F: Sync[F]): F[ThreadFactory] =
+      F.delay {
+        new ThreadFactory {
+          private val counter = new AtomicLong(0L)
+
+          def newThread(r: Runnable): Thread = {
+            val th = new Thread(r)
+            th.setName("linebacker-thread-" + counter.getAndIncrement.toString)
+            th.setDaemon(true)
+            th
+          }
+        }
+      }
+
+    def unboundExecutorUnsafe[F[_]: Sync](threadFactory: F[ThreadFactory]): F[ExecutorService] = {
```

Comment:
I'm not following why the `threadFactory` needs to be `F[_]` rather than just passed explicitly to this function.

Reply:
Yeah, that doesn't make sense. Expect another update soon ;)